src/Middleware/RateLimiting/src/RateLimitingApplicationBuilderExtensions.cs

```diff
@@ -29,11 +29,14 @@ public static IApplicationBuilder UseRateLimiter(this IApplicationBuilder app)
     /// <param name="app"></param>
     /// <param name="options"></param>
     /// <returns></returns>
-    public static IApplicationBuilder UseRateLimiter(this IApplicationBuilder app, RateLimiterOptions options)
+    public static IApplicationBuilder UseRateLimiter(this IApplicationBuilder app, Action<RateLimiterOptions> options)
     {
         ArgumentNullException.ThrowIfNull(app, nameof(app));
         ArgumentNullException.ThrowIfNull(options, nameof(options));
 
-        return app.UseMiddleware<RateLimitingMiddleware>(Options.Create(options));
+        var rateLimiterOptions = new RateLimiterOptions();
+        options?.Invoke(rateLimiterOptions);
```

Comment:
No need to check null
```suggestion
        options.Invoke(rateLimiterOptions);
```